filechanges/opt/musicbox/startup.sh

```diff
@@ -91,16 +91,31 @@ chmod u+s /sbin/shutdown
 
 if [ "$INI__network__wifi_network" != "" ]
 then
+
     #put wifi settings for wpa roaming
-cat >/etc/wpa.conf <<EOF
-    ctrl_interface=DIR=/var/run/wpa_supplicant GROUP=netdev
-    update_config=1
-    network={
-        ssid="$INI__network__wifi_network"
-        psk="$INI__network__wifi_password"
-        scan_ssid=1
-    }
-EOF
+    if [ "$INI__network__wifi_password" != "" ]
+    then
+        cat >/etc/wpa.conf <<EOF
+            ctrl_interface=DIR=/var/run/wpa_supplicant GROUP=netdev
+            update_config=1
+            network={
+                ssid="$INI__network__wifi_network"
+                psk="$INI__network__wifi_password"
+                scan_ssid=1
+            }
+        EOF
+    else
+        cat >/etc/wpa.conf <<EOF
+            ctrl_interface=DIR=/var/run/wpa_supplicant GROUP=netdev
+            update_config=1
+            network={
+                ssid="$INI__network__wifi_network"
+                proto=RSN
```

Comment:
I need this to be able to connect to an open wifi network at work. The wifi security is web signon, not WEP og WPA. 
After booting MusicBox i log on to the network using a curl command.
I guess other users also would like to logon to open network, ie. when you use MAC authentication.